data_structures/src/data_request.rs

```diff
@@ -424,57 +435,83 @@ where
             .into());
         }
 
-        let liars_count = liars.iter().fold(0, |count, liar| match liar {
-            true => count + 1,
-            false => count,
-        });
-        let honests_count = reveals_count - liars_count;
+        if reveals_count != errors.len() {
+            return Err(TransactionError::MismatchingErrorsNumber {
+                reveals_n: reveals_count,
+                inputs_n: errors.len(),
+            }
+            .into());
+        }
+
+        let (liars_count, errors_count) = calculate_errors_and_liars_count(errors, liars);
+
+        let collateral = if dr_output.collateral == 0 {
+            collateral_minimum
+        } else {
+            dr_output.collateral
+        };
```

Comment:
That it is a different approach. We consider that if there are not a collateral value, we put the collateral minimum. But if the requester specify a quantity for collateral, and is less than minimum I think we shouldnt change and wait for the data request error in validation. 

Maybe someone put 50_000 because he thinks is in wits and need a big amount of collateral, it is better that the request will be never included, and the requester notice his own error than publish the request with a very low value of collateral